src/main/java/edu/whimc/observations/observetemplate/gui/TemplateSelection.java

```diff
@@ -203,7 +203,7 @@ private void doConfirm() {
         // Create observation object for custom event
         int days = plugin.getConfig().getInt("expiration-days");
         Timestamp expiration = Timestamp.from(Instant.now().plus(days, ChronoUnit.DAYS));
-        Observation obs = Observation.createObservation(plugin, player, player.getLocation(), filledIn, expiration, null);
+        Observation obs = Observation.createObservation(plugin, player, player.getLocation(), filledIn, expiration, this.template);
```

Comment:
I'd prefer this logic only be implemented once in a static helper function inside `Observation.java`. [This](https://github.com/whimc/Observations/blob/0805ac76a2ae7db874737d74f582b913aaba2109/src/main/java/edu/whimc/observations/commands/ObserveCommand.java#L34-L46) should be moved to `Observation.java` and refactored to accept a template for the observation. The method name should be changed to be indicative that it uses the current time and perhaps also that it triggers the Bukkit event.

Reply:
Also, this is where the duplicated logic is coming from. The observation creation is implicitly handled in the call of `sendFooter` since the `withConfirm` parameter is `true`. [Lines 202-210](https://github.com/whimc/Observations/blob/7b47fcca90e57b6eff9ee80c08e994b38c7f5c8b/src/main/java/edu/whimc/observations/observetemplate/gui/TemplateSelection.java#L202-L210) can be removed while [lines 253-257](https://github.com/whimc/Observations/blob/7b47fcca90e57b6eff9ee80c08e994b38c7f5c8b/src/main/java/edu/whimc/observations/observetemplate/gui/TemplateSelection.java#L253-L257) should be refactored to use the new helper method mentioned above.